### DISCUSSION TITLE
Suddenly seeing weird negative prices returned for the 5 min blocks ?

### DISCUSSION THREAD

**Discussion by craigcurtin-dev** [amberelectric/public-api#58] [General]
Hey Myles.

In Sydney and just noticed this morning that my 5 min dashboard is suddenly returning negative values whilst the 30 minutes prices are still correct - has something changed on your end ?

(I have not made any changes to my production scripts (Node-Red) or the server it runs on in over a week)

Craig

**Comment by madpilot** [comment]
Can you show me a screenshot/output?

**Reply by craigcurtin-dev** [reply]
Here you go Myles,

![image](https://user-images.githubusercontent.com/54323535/127950228-dc890544-a910-44d8-bccc-3ce0e32203fe.png)

Basically i have a routine that pulls out the 5 min prices at 1 min past the hour onwards (so give you time to process the AEMO feed as the 5min comes in)

This has been working fine for about the last 4 weeks or so since i put it in place

And Just noticed this morning the change.

If you have not changed anything at your end i will dig into my code and do some debugging

Craig

**Reply by craigcurtin-dev** [reply]
And here is the Node Red Debug window of the latest call

![image](https://user-images.githubusercontent.com/54323535/127950968-c3642bad-ddce-468f-8bdb-049e4cc2e9a9.png)

And here is the API call i am making

https://api.amber.com.au/v1/sites/01EC780K3J6E1SFKAH5SCQAZ0F/prices/current?resolution=30

Craig

**Reply by craigcurtin-dev** [reply]
I don't want to seem stupid - but have you suddenly reversed the order of the information returned ?

My code all expects their to be two objects returned - the feedin and the general - it looks like their order has been reversed ?

Craig

**Reply by madpilot** [reply]
The order isn't guaranteed, it's whatever order they come out of the market meter database - we did a big tariff update yesterday, so it's entirely possible the order changed.

I'm going to put a card to fix the ordering - you are the third person this has bitten.

**Reply by madpilot** [reply]
If you are now using feed in as the source, that would explain the negative numbers.

**Reply by craigcurtin-dev** [reply]
Yep indeed it would - i just took the two returned sets of values and but their in javascript objects and then assumed that i was getting the correct price.

So should i fix my code to look for feedin and general (is that the correct way to identify them ?) or will you revert at your end ?

Craig

**Reply by oldwindmill** [reply]
i found the order of general and feedin reversed a few weeks ago. despite Myles warning the order was not guaranteed...maybe it flipped back...

**Reply by oldwindmill** [reply]
i updated my code to check i was working on the correct value. i think that is the right approach.

**Reply by madpilot** [reply]
I'm just pushing a fix now - should be available in about 30 minutes.

**Reply by madpilot** [reply]
@oldwindmill @craigcurtin-dev The order was non-deterministic per user, so the order you were seeing it, isn't the order another user is seeing it.

Once this patch goes live it will always be general > controlled load > feed In

**Comment by craigcurtin-dev** [comment]
OK no worries – I have updated my lazy piece of coding and put in a JSONATA substitution to move the general prices into my main JSO now

Thanks for the quick reply

Craig

**Comment by madpilot** [comment]
That fix is in prod now.

**Comment by craigcurtin-dev** [comment]
I wish there were more developers around like you !! Willing to listen and act !

Craig
